generators/exercises/list_ops.py

```diff
@@ -0,0 +1,124 @@
+extern size_t reverse(int64_t *buffer, const int64_t *list, size_t list_count);
+
+static bool is_odd(int64_t x) {
+    return x % 2 == 1;
+}
+
+static int64_t plus_one(int64_t x) {
+    return x + 1;
+}
+
+static int64_t multiply(int64_t acc, int64_t el) {
+    return el * acc;
+}
+
+static int64_t add(int64_t acc, int64_t el) {
+    return el + acc;
+}
+
+static int64_t divide(int64_t acc, int64_t el) {
+    return acc == 0 ? 0 : el / acc;
+}
+"""
+
+def array_literal(digits):
+    return '{' + ', '.join(str(d) for d in digits) + '}'
+
+def describe(case):
+    description = case["description"]
+    property = case["property"]
+    return f"{property} {description}"
+
+def gen_func_body(prop, inp, expected):
+    str_list = []
+
+    if prop in ['foldl', 'foldr']:
+        initial = inp['initial']
+
+    function = None
+    if prop not in ['append', 'reverse']:
+        function = inp['function']
+        if function == "(x) -> x modulo 2 == 1":
+            function = 'is_odd'
+        if function == "(x) -> x + 1":
+            function = 'plus_one'
+
+        if function == "(acc, el) -> el * acc":
+            function = 'multiply'
+        if function == "(acc, el) -> el + acc":
+            function = 'add'
+        if function == "(acc, el) -> el / acc":
+            function = 'divide'
+
+    if prop == 'append':
+        list1 = inp['list1']
+        list2 = inp['list2']
+        if len(list1) > 0:
+            str_list.append(f'const int64_t list1[] = {array_literal(list1)};\n')
+            list1_with_count = 'list1, ARRAY_SIZE(list1)'
+        else:
+            list1_with_count = 'NULL, 0'
+        if len(list2) > 0:
+            str_list.append(f'const int64_t list2[] = {array_literal(list2)};\n')
+            list2_with_count = 'list2, ARRAY_SIZE(list2)'
+        else:
+            list2_with_count = 'NULL, 0'
+    else:
+        list_ = inp['list']
+        if function == 'divide':
+            initial = 5
+            list_ = [2, 5]
+            if prop == 'foldl':
+                expected = 0
+            else:
+                expected = 2
+
+        if len(list_) > 0:
+            str_list.append(f'const int64_t list[] = {array_literal(list_)};\n')
+            list_with_count = 'list, ARRAY_SIZE(list)'
+        else:
+            list_with_count = 'NULL, 0'
+
+    if prop == 'append':
+        str_list.append('int64_t buffer[BUFFER_SIZE];\n')
+        call = f'{prop}(buffer, {list1_with_count}, {list2_with_count})'
+
+    if prop in ['filter', 'map']:
+        str_list.append('int64_t buffer[BUFFER_SIZE];\n')
+        call = f'{prop}(buffer, {list_with_count}, {function})'
+
+    if prop in ['foldl', 'foldr']:
+        call = f'{prop}({list_with_count}, {initial}, {function})'
+
+    if prop == 'reverse':
+        str_list.append('int64_t buffer[BUFFER_SIZE];\n')
+        call = f'{prop}(buffer, {list_with_count})'
```

Comment:
You should use `elif` to avoid evaluations that for sure will be false. Or use pattern matching:
```suggestion
    match prop:
        case 'append':
            str_list.append('int64_t buffer[BUFFER_SIZE];\n')
            call = f'{prop}(buffer, {list1_with_count}, {list2_with_count})'
        case 'filter' | 'map':
            str_list.append('int64_t buffer[BUFFER_SIZE];\n')
            call = f'{prop}(buffer, {list_with_count}, {function})'
        case 'foldl' | 'foldr':
            call = f'{prop}({list_with_count}, {initial}, {function})'
        case 'reverse':
            str_list.append('int64_t buffer[BUFFER_SIZE];\n')
            call = f'{prop}(buffer, {list_with_count})'
```